source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/input/ChoiceSetInputRenderer.java

```diff
@@ -612,6 +627,117 @@ public void onNothingSelected(AdapterView<?> parent)
         }
     }
 
+    public View renderTypeAheadControl(
+        RenderedAdaptiveCard renderedCard,
+        Context context,
+        ChoiceSetInput choiceSetInput,
+        IChoicesResolver choicesResolver,
+        HostConfig hostConfig,
+        RenderArgs renderArgs)
+    {
+        final List<String> titleList = new ArrayList();
+        ChoiceInputVector choiceInputVector = choiceSetInput.GetChoices();
+        long size = choiceInputVector.size();
+        int valueIndex = -1;
+        String value = choiceSetInput.GetValue();
+        final List<String> valueList = new ArrayList();
+        for (int i = 0; i < size; i++)
+        {
+            ChoiceInput choiceInput = choiceInputVector.get(i);
+            titleList.add(choiceInput.GetTitle());
+            valueList.add(choiceInput.GetTitle());
+
+            if (choiceInput.GetValue().equals(value))
+            {
+                valueIndex = i;
+            }
+        }
+
+        boolean usingCustomInputs = isUsingCustomInputs(context);
+
+        ValidatedTextView validatedTypeAheadTextView = new ValidatedTextView(context, usingCustomInputs);
+
+        Drawable mDrawable = getDrawable(context, R.drawable.adaptive_card_ic_chevron_right);
+        validatedTypeAheadTextView.setCompoundDrawablesWithIntrinsicBounds(null, null, mDrawable, null);
+        validatedTypeAheadTextView.setPaddingRelative(0, 10, 20, 10);
+        validatedTypeAheadTextView.setEllipsize(TextUtils.TruncateAt.END);
+
+        final TypeAheadTextViewHandler typeAheadTextInputHandler = new TypeAheadTextViewHandler(choiceSetInput);
+
+        ValidatedInputLayout inputLayout = null;
+
+        // if using custom inputs, we don't have to create the surrounding linear layout
+        boolean needsOuterLayout = (!usingCustomInputs);
```

Comment:
this part seems like boilerplate code that may be shared with filteredChoiceSet, can you open a bug for refactoring this?

Reply:
sure